tests/vmi_networking_test.go

```diff
@@ -863,3 +860,50 @@ func NewRandomVMIWithInvalidNetworkInterface() *v1.VirtualMachineInstance {
 	vmi.Spec.Domain.Devices.Interfaces[0].Model = "gibberish"
 	return vmi
 }
+
+func createExpectTraceroute6(address string) []expect.Batcher {
+	return []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: "traceroute -6 " + address + " -w1 > tr\n"},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: "cat tr | grep -q \"*\\|!\"\n"},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: "echo $?\n"},
+		&expect.BExp{R: "1"},
+	}
+}
+
+func createExpectStartTcpServer(port string) []expect.Batcher {
+	return []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: "screen -d -m sudo nc -klp " + port + " -e echo -e 'Hello World!'\n"},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: "echo $?\n"},
+		&expect.BExp{R: "0"},
+	}
+}
+
+func createExpectConnectToServer(serverIP, tcpPort string, expectSuccess bool) []expect.Batcher {
+	expectResult := "1"
+	if expectSuccess {
+		expectResult = "0"
+	}
+	return []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: fmt.Sprintf("echo test | nc %s %s -i 1 -w 1 1> /dev/null\n", serverIP, tcpPort)},
+		&expect.BExp{R: "\\$ "},
+		&expect.BSnd{S: "echo $?\n"},
+		&expect.BExp{R: expectResult},
+	}
+}
+
+// gatewayIpFromCIDR returns the first address of a network.
+func gatewayIPFromCIDR(cidr string) string {
```

Comment:
You can do something similar to `[GetHostAndGwAddressesFromCIDR`](https://github.com/kubevirt/kubevirt/blob/51ce9d8e6def33c0c260f0cdb994f08ae5b9938e/pkg/virt-launcher/virtwrap/network/common.go#L218) so it will works for both ipv4 and ipv6.

Reply:
My final version with IPv6 support for these tests was this one:
```
func gatewayIPFromCIDR(cidr string) string {
	ip, _, _ := net.ParseCIDR(cidr)
	if netutils.IsIPv6(ip) {
		ip = ip.To16()
		ip[15]++
	} else {
		ip = ip.To4()
		ip[3]++
	}
	return ip.String()
```

In order to use the one you mentioned I'll need to move it to some package that can be shared by all (production and tests). Copying it here seems an overkill, as we do not need to support all types of input.
Are you ok with the version mentioned here (will be added with IPv6 support) or you prefer to use the big one?

Reply:
Although, if I take only snippets from that method, it will look like this:
Seems nicer.

```
	ip, net, _ := net.ParseCIDR(cidr)
	ip = ip.Mask(net.Mask)
	oct := len(ip) - 1
	ip[oct]++
	return ip.String()
```